File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/domain/repositories/RepositoriesDomainContract.java
package com.duxtinto.intellij.plugin.github.codereviews.domain.repositories;

import com.duxtinto.intellij.plugin.github.codereviews.ide.acl.entities.GitRepositoryExt;

import javax.annotation.Nullable;

public interface RepositoriesDomainContract {
    interface GithubRepositoryFinder {
        @Nullable
        GithubRepositoryEntity find(GitRepositoryExt gitRepository);
    }

    interface GitRepositoryFinder {
        @Nullable
        GitRepositoryExt findRootRepo();
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/domain/repositories/RepositoriesDomainContract.kt
package com.duxtinto.intellij.plugin.github.codereviews.domain.repositories

interface RepositoriesDomainContract {
    interface GithubRepositoryFinder {
        fun find(gitRepository: Git.Repository): GithubRepositoryEntity?
    }

    interface Git {
        interface Repository {
            fun hasGithubRemote() : Boolean
        }

        interface RepositoryFinder {
            fun findRootRepo(): Repository?
        }

        interface BranchOperator {
            fun checkOut(ref: String, repository: Git.Repository)
        }
    }
}
